mesonbuild/compilers/c.py

```diff
@@ -288,6 +288,7 @@ def has_header_symbol(self, hname, symbol, prefix, env, extra_args=None, depende
             #ifndef {symbol}
                 {symbol};
             #endif
+            return 0;
```

Comment:
Setting CFLAGS in the environment does not cause those flags to be added to compiler checks. Are you sure this is needed?

Reply:
It certainly seemed to cause GLib to fail to build when I tested it quickly.

Reply:
I was wrong, but we should probably also add `['-Wno-all', '-Wno-error']` to `compilers/c.py:get_compiler_check_args()` because some of our tests do funky stuff that will never pass through a strict compiler, f.ex. `has_function()`.

See `compilers/c.py:_get_compiler_check_args()` (note the underscore in this one) for a list of all source from which compiler arguments are gathered for compiler checks.

Reply:
... and we should also add a unit test for this to ensure that passing `-Werror` in CFLAGS does not cause a compiler check that is known to error out with it to fail.

Reply:
What’s the status of this now that #3720 is fixed?